lib/sycamore/sycamore/docset.py

```diff
@@ -1531,29 +1530,45 @@ def execute(self, **kwargs) -> None:
         from sycamore.executor import Execution
         from sycamore.materialize import Materialize
 
-        if isinstance(self.plan, Materialize) and self.plan._reliability is not None:
-            mrr = self.plan._reliability
-
-            assert self.plan.child().last_node_materialize(), "first step must be materialize step"
+        if isinstance(self.plan, Materialize):
+            mrr = None
+            for rule in self.context.rewrite_rules:
+                if isinstance(rule, MaterializeReadReliability):
+                    mrr = rule
+                    if isinstance(self.plan._orig_path, str):
```

Comment:
Check to make sure this works with S3 & non S3 paths. I remember some weirdness there.
Single manual test is sufficient.